### PR TITLE
multistage: handle imagestream references in From

### PR DESCRIPTION
**Pull request by vrutkovs**
Resubmit of https://github.com/openshift/ci-tools/pull/1230 (reverted by https://github.com/openshift/ci-tools/pull/1238).

If multistage "from" contains imagestreamtag it would be used in the pod. This is useful for upgrade jobs, which require installer to be used from "stable-initial:installer" imagstreamtag.

### CODE REVIEW

**Comment by bbguimaraes** [comment]
The reason the validation existed in the first place was that the origin of an image could be determined from the configuration — any missing name was considered to be imported into `stable`, which we cannot validate before importing the tags.  The origin is required not only to determine the value for the Pod's `image`, but also for dependency resolution.

Since the customization is only required when multiple releases are imported, can we limit it to `stable-<name>:<image>`?  That way we keep the ability to correctly identify the source and also support multiple releases.

@stevekuznetsov

/hold

**Comment by vrutkovs** [comment]
>Since the customization is only required when multiple releases are imported, can we limit it to stable-<name>:<image>? That way we keep the ability to correctly identify the source and also support multiple releases.

Added validation of allowed imagestreams: `pipeline`/`stable`/`stable-initial`/`release`

File: pkg/steps/multi_stage.go [hunk 1]
```diff
@@ -353,8 +353,11 @@ func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []cor
 		if link, ok := step.FromImageTag(); ok {
 			image = fmt.Sprintf("%s:%s", api.PipelineImageStream, link)
 		} else {
-			stream, _ := s.config.ImageStreamFor(image)
-			image = fmt.Sprintf("%s:%s", stream, image)
+			parts := strings.Split(image, ":")
```
Comment (stevekuznetsov) [inline pkg/steps/multi_stage.go:356]: You want `DependencyParts` here - see other usage in the repo

File: pkg/steps/multi_stage.go [hunk 2]
```diff
@@ -353,8 +353,10 @@ func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []cor
 		if link, ok := step.FromImageTag(); ok {
 			image = fmt.Sprintf("%s:%s", api.PipelineImageStream, link)
 		} else {
-			stream, _ := s.config.ImageStreamFor(image)
-			image = fmt.Sprintf("%s:%s", stream, image)
+			dep := api.StepDependency{Name: image}
```
Comment (stevekuznetsov) [inline pkg/steps/multi_stage.go:356]: Sorry, implicit in that statement was that you also need to expose those dependencies to the graph builder - you'll need to do this resolution and use `LinkForImage()`, expose using the step's `Requires()` method

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**